kiwi/xml_state.py

```diff
@@ -2110,6 +2108,37 @@ def get_root_filesystem_uuid(self) -> Optional[str]:
         """
         return self.root_filesystem_uuid
 
+    def get_bootstrap_archives_target_dirs(self) -> Dict:
+        """
+        Dict of archive names and target dirs from the type="bootstrap"
+        packages section(s)
+        :return: archive names and its target dir
+        :rtype: dict
+        """
+        result = {}
+        bootstrap_packages_sections = self.get_bootstrap_packages_sections()
+        if bootstrap_packages_sections:
+            for bootstrap_packages_section in bootstrap_packages_sections:
+                for archive in bootstrap_packages_section.get_archive():
+                    result[archive.get_name().strip()] = archive.get_target_dir()
```

Comment:
This could be done in `get_bootstrap_archives` (and same for `get_system_archives`) but
it would change the method, returning a `dict` not a `list` and few extra changes but 
there would be less duplication and also this loop would be done fewer times.

I was not sure about changing those methods but if everybody is ok, more than happy 
to do the update

Reply:
yeah I was thinking the same but then realized these are methods of the public API and we cannot change their return value type unless we do a major version upgrade of kiwi. So the way you designed it is great, let's keep that.

However, on the code duplication end the implementation could be improved imho. You can think of something like this:

```python
def get_archives_target_dirs(self, package_section_names: List[str]) -> Dict:
    ...

def get_system_archives_target_dirs(self) -> Dict:
    return self.get_archives_target_dirs(
        self.get_packages_sections(['image', self.get_build_type_name()])
    ) 

def get_bootstrap_archives_target_dirs(self) -> Dict:
    return self.get_archives_target_dirs(
        self.get_packages_sections(['bootstrap'])
    ) 
```

Reply:
Also remember that list types in generateDS created data structures can return None instead of []. This should be taken care in the implementation of ```get_archives_target_dirs()```